### PR TITLE
[Group] Yaml test files for group messaging demonstration

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Problem
* Chip-tool cannot send group messages yet.

#### Change overview
* Add yaml tests to simplify testing group messaging
* No functionnal changes

#### Testing
* Manual test with efr32 to validate that two configured devices can receive group messages

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking, given this is adding tests.

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -20,8 +20,7 @@ on:
     workflow_dispatch:
 
 concurrency:
-    group:
```
Comment (bzbarsky-apple) [inline .github/workflows/tests.yaml:23]: Why the random reformatting throughout this file?

File: examples/chip-tool/templates/tests.js [hunk 2]
```diff
@@ -279,6 +284,7 @@ function getTests()
     Others, //
     SoftwareDiagnostics, //
     Subscriptions, //
+    Groups,
```
Comment (bzbarsky-apple) [inline examples/chip-tool/templates/tests.js:287]:
```suggestion
    Groups, //
```
to keep the reformatter at bay

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -88,6 +84,7 @@ jobs:
               run: |
                   ./scripts/run_in_build_env.sh \
                   "./scripts/tests/run_test_suite.py \
+                     --target-skip-glob '{TestGroupMessaging, TestGroupDemoCommand, TestGroupDemoConfig, TV_*}' \
```
Comment (bzbarsky-apple) [inline .github/workflows/tests.yaml:87]: Why are we disabling all the TV tests here now?  This does not look right, and all the whitespace noise was covering it up.

File: src/app/tests/suites/TestGroupDemoCommand.yaml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# !!!!!!!!!! DO NOT RUN IN CI - ONLY USED IN MANUAL GROUP MESSAGING DEMONSTRATION !!!!!!!!!!
+
+# This test files is used in the group messaging demonstration to send on/off commands to light-apps that have been added to the group with TestGroupDemoConfig
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupDemoCommand.yaml:17]:
```suggestion
# This test file is used in the group messaging demonstration to send on/off commands to light-apps that have been added to the group with TestGroupDemoConfig
```

File: src/app/tests/suites/TestGroupDemoCommand.yaml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+# This file is used in tadem with TestGroupDemoConfig that configures the light-apps.
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupDemoCommand.yaml:18]:
```suggestion
# This file is used in tandem with TestGroupDemoConfig that configures the light-apps.
```

File: src/app/tests/suites/TestGroupDemoCommand.yaml [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+#
+# To run this test file : chip-tool tests TestGroupDemoCommand <nodeid>
+#   The nodeid isn't use in the actual tests, but it is necessary with the current test infrastructure.
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupDemoCommand.yaml:21]:
```suggestion
#   The nodeid isn't used in the actual tests, but it is necessary with the current test infrastructure.
```

File: src/app/tests/suites/TestGroupDemoCommand.yaml [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+#   Use the node id of one the commisionned light-apps.
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupDemoCommand.yaml:22]:
```suggestion
#   Use the node id of one of the commisioned light-apps.
```

File: src/app/tests/suites/TestGroupDemoCommand.yaml [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+#
+# Full test procedure :
+#   chip-tool tests TestGroupDemoConfig <nodeid>  - Run this command every light app nodeid that will be used for the demonstration
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupDemoCommand.yaml:25]:
```suggestion
#   chip-tool tests TestGroupDemoConfig <nodeid>  - Run this command for every light app nodeid that will be used for the demonstration
```

File: src/app/tests/suites/TestGroupDemoConfig.yaml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# !!!!!!!!!! DO NOT RUN IN CI - ONLY USED IN MANUAL GROUP MESSAGING DEMONSTRATION !!!!!!!!!!
+
+# This test files is used in the group messaging demonstration to configure light-apps
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupDemoConfig.yaml:17]:
```suggestion
# This test file is used in the group messaging demonstration to configure light-apps
```

File: src/app/tests/suites/TestGroupDemoConfig.yaml [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+# Run this test on a commisionned device to add it to the group.
+# This file is used in tadem with TestGroupDemoCommand that sends the actual commands.
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupDemoConfig.yaml:19]:
```suggestion
# This file is used in tandem with TestGroupDemoCommand that sends the actual commands.
```

File: src/app/tests/suites/TestGroupDemoConfig.yaml [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+#
+# To run this test file : chip-tool tests TestGroupDemoConfig <nodeid>
+#
+# Full test procedure :
+#   chip-tool tests TestGroupDemoConfig <nodeid>  - Run this command every light app nodeid that will be used for the demonstration
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestGroupDemoConfig.yaml:24]:
```suggestion
#   chip-tool tests TestGroupDemoConfig <nodeid>  - Run this command for every light app nodeid that will be used for the demonstration
```

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yaml (hunk 3 above)
Comment (mkardous-silabs) [reply .github/workflows/tests.yaml:87]: The TV tests were already disabled for darwin. The only thing that was added in this PR is TestGroupDemoCommand and TestGroupDemoConfig.

Comment (mkardous-silabs) [reply .github/workflows/tests.yaml:87]: copy / paste mistake, i correted it.

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (mkardous-silabs) [reply .github/workflows/tests.yaml]: I assumed it was the restyle that did it. I removed all the style changes.

File: .github/workflows/tests.yaml [hunk 12]
```diff
@@ -178,7 +179,7 @@ jobs:
                   ./scripts/run_in_build_env.sh \
                   "./scripts/tests/run_test_suite.py \
                      --chip-tool ./out/darwin-x64-chip-tool-${BUILD_VARIANT}${CHIP_TOOL_VARIANT}/chip-tool \
-                     --target-skip-glob '{TestGroupMessaging,TV_*}' \
+                     --target-skip-glob '{TestGroupMessaging,TestGroupDemoCommand,TestGroupDemoConfig,TV_*}' \
```
Comment (andy31415) [inline .github/workflows/tests.yaml:182]: should the skip glob be `TestGroup*` ? Or are there tests included with that prefix that pass?

File: .github/workflows/tests.yaml [hunk 13]
```diff
@@ -88,6 +88,7 @@ jobs:
               run: |
                   ./scripts/run_in_build_env.sh \
                   "./scripts/tests/run_test_suite.py \
+                     --target-skip-glob '{TestGroupDemoCommand,TestGroupDemoConfig}' \
```
Comment (andy31415) [inline .github/workflows/tests.yaml:91]: Please add some comment explaining why these demo tests are globbed out of CI everywhere.

If CI does not execute any of these tests, the tests will easily get broken in time

**Review by bzbarsky-apple — APPROVED** [review]

File: .github/workflows/tests.yaml (hunk 12 above)
Comment (mkardous-silabs) [reply .github/workflows/tests.yaml:182]: No, other tests need to be run on darwin like `TestGroupsCluster` and `TestGroupKeyManagementCluster`.

File: .github/workflows/tests.yaml (hunk 13 above)
Comment (mkardous-silabs) [reply .github/workflows/tests.yaml:91]: In each of the test files, there is an explanation on why it shouldn't be run by the CI. Both test files are only used for manual group tests to simplify testing for manifacturers. The actual group messaging tests are done by `TestGroupMessaging`.

**Pull request merged**